ondir.c

```diff
@@ -77,7 +77,7 @@ const char *src = NULL, *dst = NULL, *home = NULL;
 	len = strlen(working);
 
 	/* Traverse up source path */
-	while (strncmp(working, dst, len)) {
+	while (strncmp(working, dst, len) || (dst[len] && dst[len] != '/')) {
```

Comment:
This one is a one line change, and should never be an access violation so long as dst is null terminated. Not positive about the `dst[len] != '/'` check, it might be enough just to check that `dst[len]` is not null.